### DISCUSSION TITLE
CORS errors with custom header in route

### DISCUSSION THREAD

**Discussion by priyanka-raghavan-maersk** [Kong/kong#8647] [Help] [answered]
Hi Everyone,
I am using the Kong oss for our gateway and i am seeing this cors issue when we have a custom header.

THE DOCUMENTATIONS SAYS THIS
_**If the client is a browser, there is a known issue with this plugin caused by a limitation of the CORS specification that doesn’t allow to specify a custom Host header in a preflight OPTIONS request.
Because of this limitation, this plugin only works for Routes that have been configured with a paths setting. The CORS plugin does not work for Routes that are being resolved using a custom DNS (the hosts property).**_

I do have the route only set with path and not host. Any ideas why it is still not working?  I followed what was suggested here but still does not work https://github.com/Kong/kong/issues/2027

Appreciate any help?

**Comment by kikito** [comment]
Hello Priyanka,

It is difficult to know exactly what is going on with what you are sharing. Could you please add more details to your question?
* Which version of Kong are you using?
* Are you using any other plugins, especially any custom ones?
* How are you configuring your Routes, Services and Plugin, exactly (you may blank private information)
* An example of a request that you are making, and the response that you got (hopefully showing that the CORS plugin does not work)
* Any errors that might have popped in the Kong log

**Comment by priyanka-raghavan-maersk** [comment, marked as answer]
Hello @kikito ,

Sorry the version we are using kong 2.6.0 and we do have some custom plugins.
We are using the admin API and deck commands to make changes.

I do not see anything of significance in the docker logs

Which version of Kong are you using?-->2.6
Are you using any other plugins, especially any custom ones?-->YES WE have two custom plugins
How are you configuring your Routes, Services and Plugin, exactly (you may blank private information)
Route information
![image](https://user-images.githubusercontent.com/82929831/162361591-36d48439-fae0-4ac3-a004-28552fca8ba1.png)
In the cors plugin for access-control-allow-headers, i added this custom header apiversion 

The request I am making through UI is like this
I am able to replicate this also in custom UI on my local machine

 var myHeaders = new Headers();
    myHeaders.append("Service-Key", "servicekey");
    myHeaders.append("apiversion", "2");
 var requestOptions = {
        method: 'GET',
        headers: myHeaders
    };
  let response = await fetch(url,requestOptions);
 Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.


As soon as i remove the custom header apiversion from the route in Kong. Everything works.

**Reply by priyanka-raghavan-maersk** [reply]
Issue is only from browser. Is there a way to avoid making a preflight request?

**Reply by priyanka-raghavan-maersk** [reply]
We created a custom plugin and all seems quiet on the western front now.
